apps/dsb-client-gateway-api/src/app/modules/message/service/message.service.ts

```diff
@@ -507,14 +553,16 @@ export class MessageService {
       `[getMessages] Total fulfilled messages ${messageResponses.length}`
     );
 
-    if (ack) {
-      const successAckMessageIds: string[] = await this.sendAckBy(
-        fulfilledMessages.map((message) => message.id),
-        `${clientId}:${fqcn}`
-      );
-      fulfilledMessages = fulfilledMessages.filter((msg) =>
-        successAckMessageIds.includes(msg.id)
-      );
+    const idsPendingAck: PendingAcksEntity[] = fulfilledMessages.map(e => {
+      return {
+        clientId: consumer,
+        messageId: e.id,
+        mbTimestamp: moment(e.timestampNanos / (1000 * 1000)).utc().toDate()
+      }
+    });
+
+    if (idsPendingAck.length > 0) {
+      this.pendingAcksWrapperRepository.pendingAcksRepository.save(idsPendingAck).then();
```

Comment:
missing await -> this may be executed after some time if event loop is full

Reply:
using then()